internal/graph/memory_test.go

```diff
@@ -0,0 +1,412 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package graph
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"reflect"
+	"testing"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/internal/cas"
+	"oras.land/oras-go/v2/internal/descriptor"
+)
+
+/*
+A(manifest)-------------------+
+
+	|                         |
+	|                         |
+	|                         |
+	v                         |
+
+B(manifest)-----------------+ |
+
+	|                       | |
+	|                       | |
+	v                       v v
+
+C(layer)                  D(layer)
+*/
+func TestDeletableMemory_IndexAndRemove(t *testing.T) {
+	testFetcher := cas.NewMemory()
```

Comment:
nit: It's better to call it a `storage` instead of a `fetcher`, as we invoke both `Push` and `Fetch` on it.

Reply:
I think `testFetcher` is a better name, since `Index()` requires a `Fetcher` argument, and the sole purpose of this variable is to test `Index`. We only `Push` to it so that the `Fetcher` has some content in it.